### ISSUE TITLE
Error serializing DataStore data with arrays of custom types: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Not a JSON Object: []

### ISSUE THREAD

**Issue by Geol1** [aws-amplify/amplify-flutter#3482]
### Description

**_I have an error when I add a table with a custom type. Whenever I try to create an Info by filling in the comment and asset fields, whether it's an empty array ([]) or an array of values, it gives me an error
Everything works fine with DataStore when comments and assets are defined as [String] or [AWSJSON] but I still have [Comment] and [Asset] to keep commits in my data_**

I/amplify:aws-datastore(26048): Successfully enqueued PendingMutation{mutatedItem=SerializedModel{id='2208902c-e9f2-48e6-8f02-c589554a7c78', serializedData={code=info_DIVE-info-user_ba1e703e-56f8-4c97-93ab-632f18, comments=[], publishedAt=2023-07-31T06:55:25.003089000Z, rating=0, description=_description.text, endAt=2023-07-31T06:55:25.003089000Z, title=title.text, type=DIVERS, url=, tags=[], createdAt=null, assets=[SerializedCustomType{serializedData={number=0, path=info-images/1690786521196-scaled_IMG_20220328_132837.jpg, size=172296.0, name=scaled_IMG_20220328_132837.jpg, id=53457e49-d579-4eef-b818-d43a9c46ca7e, contentType=PICTURE, content=null}, customTypeName=CustomTypeSchema{name='Asset', pluralName='Assets', fields={content=CustomTypeField{name='content', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isCustomType=false}, contentType=CustomTypeField{name='contentType', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isCustomType=false}, id=CustomTypeField{name='id', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isCustomType=false}, name=CustomTypeField{name='name', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isCustomType=false}, number=CustomTypeField{name='number', javaClassForValue='class java.lang.Integer', targetType='Integer', isRequired=false, isArray=false, isEnum=false, isCustomType=false}, path=CustomTypeField{name='path', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isCustomType=false}, size=CustomTypeField{name='size', javaClassForValue='class java.lang.Double', targetType='Double', isRequired=false, isArray=false, isEnum=false, isCustomType=false}}}}], alert=[], id=2208902c-e9f2-48e6-8f02-c589554a7c78, user=SerializedModel{id='ba1e703e-56f8-4c97-93ab-632f18e5efe1', serializedData={id=ba1e703e-56f8-4c97-93ab-632f18e5efe1}, modelName=User}, views=0, likes=[], updatedAt=null}, modelName=Info}, mutationType=CREATE, mutationId=39b9d832-2f6f-11ee-b377-710ac5d76df0, predicate=MatchAllQueryPredicate}
**E/amplify:aws-datastore(26048): Error ended observation of mutation outbox:
E/amplify:aws-datastore(26048): com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Not a JSON Object: []
E/amplify:aws-datastore(26048): 	at com.google.gson.Gson.fromJson(Gson.java:975)
E/amplify:aws-datastore(26048): 	at com.google.gson.Gson.fromJson(Gson.java:1034)
E/amplify:aws-datastore(26048): 	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.appsync.SerializedModelAdapter.deserialize(SerializedModelAdapter.java:115)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.appsync.SerializedModelAdapter.deserialize(SerializedModelAdapter.java:43)
E/amplify:aws-datastore(26048): 	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
E/amplify:aws-datastore(26048): 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
E/amplify:aws-datastore(26048): 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
E/amplify:aws-datastore(26048): 	at com.google.gson.Gson.fromJson(Gson.java:963)
E/amplify:aws-datastore(26048): 	at com.google.gson.Gson.fromJson(Gson.java:928)**
E/amplify:aws-datastore(26048): 	at com.google.gson.Gson.fromJson(Gson.java:877)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.syncengine.GsonPendingMutationConverter.fromRecord(GsonPendingMutationConverter.java:72)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.syncengine.PersistentMutationOutbox.lambda$load$15$com-amplifyframework-datastore-syncengine-PersistentMutationOutbox(PersistentMutationOutbox.java:260)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.syncengine.PersistentMutationOutbox$$ExternalSyntheticLambda16.accept(Unknown Source:6)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$query$4$com-amplifyframework-datastore-storage-sqlite-SQLiteStorageAdapter(SQLiteStorageAdapter.java:406)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter$$ExternalSyntheticLambda0.run(Unknown Source:10)
E/amplify:aws-datastore(26048): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
E/amplify:aws-datastore(26048): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/amplify:aws-datastore(26048): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/amplify:aws-datastore(26048): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/amplify:aws-datastore(26048): 	at java.lang.Thread.run(Thread.java:929)
**E/amplify:aws-datastore(26048): Caused by: java.lang.IllegalStateException: Not a JSON Object: []
E/amplify:aws-datastore(26048): 	at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:91)
E/amplify:aws-datastore(26048): 	at** com.amplifyframework.datastore.appsync.SerializedCustomTypeAdapter.deserialize(SerializedCustomTypeAdapter.java:80)
E/amplify:aws-datastore(26048): 	at com.amplifyframework.datastore.appsync.SerializedCustomTypeAdapter.deserialize(SerializedCustomTypeAdapter.java:41)
E/amplify:aws-datastore(26048): 	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
E/amplify:aws-datastore(26048): 	at com.google.gson.Gson.fromJson(Gson.java:963)
E/amplify:aws-datastore(26048): 	... 20 more

My code: 
User user = await _globalFunction.getConnectedUser();
      List<Asset> assets = [];
      var key;
      List<Future<void>> uploadTasks = [];

      // Upload image to Storage
      for (final imageFile in fileList!) {
        final size = await imageFile?.length();
        key =
            'info-images/${DateTime.now().millisecondsSinceEpoch}';

        // Create a task for each upload operation and add it to the list
        final uploadTask = _globalFunction.uploadedFile(
          path: imageFile.path,
          file: File(imageFile.path),
          key: key,
          name: imageFile.name,
        );
        uploadTasks.add(uploadTask);

        final asset = Asset(
            id: uuid(),
            name: imageFile.name,
            number: 0,
            size: size?.toDouble(),
            path: key,
            content: imageFile.mimeType,
            contentType: ContentType.PICTURE);

        assets.add(asset);
      }

      // Wait for all the upload tasks to complete
      await Future.wait(uploadTasks);

      TemporalDateTime dateTime = TemporalDateTime(DateTime.now());

      // Create Info
      Info inf = Info(
        title: "title.text",
        user: user,
        comments: [],
        assets: assets,
      );
      await Amplify.DataStore.save(inf);

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

1. Create amplify project
2.  Add schema in the description with the Custom type Asset and Comment
3. Build your flutter app and try to create info with a value of a field assets and comment

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.6

### Amplify Flutter Version

1.2.1

### Deployment Method

Amplify CLI

### Schema

```GraphQL
type User
  @model
  @auth(
    rules: [
      { allow: public, operations: [create, read, update], provider: apiKey }
      { allow: private, operations: [create, read, update] }
    ]
  ) {
  id: ID! 
  ...
  infos: [Info] @hasMany(indexName: "byUser", fields: ["id"]) 
}

type Info
  @model
  @auth(
    rules: [
      { allow: public, operations: [create, read, update], provider: apiKey }
      { allow: private, operations: [create, read, update] }
    ]
  ) {
  id: ID! 
  title: String! 
  type: InfoType! 
  assets: [Asset]
  rating: Int 
  likes: [String]
  comments: [Comment] 
  userID: ID! @index(name: "byUser", sortKeyFields: ["title"]) 
  user: User @belongsTo(fields: ["userID"]) 
}
type Comment {
  id: ID! # Identifiant du commentaire
  picture: String # Lien vers une image associée au commentaire
  comment: String # Contenu du commentaire
  rate: Float # Évaluation du commentaire
}

type Asset {
  id: ID! # Identifiant de la ressource
  ...
  contentType: ContentType # Type de contenu de la ressource
}

enum InfoType {
  DIVERS   AUTRE # Autres types d'informations
}
enum Status {
  ...
}

enum ContentType {
  PICTURE # Type de contenu : Image
  AUTRE
}

**Comment by dnys1** [comment]
Sorry you are facing this issue, @Geol1. We are investigating the issue in https://github.com/aws-amplify/amplify-flutter/issues/3472. Closing as a duplicate.

**Comment by abdallahshaban557** [comment]
Hello! This issue has been resolved as part of v1.3.2!
